spec/helpers/appstore_project_helper.rb

```diff
@@ -0,0 +1,221 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2018 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+# this is a duplicate of the appstore project helper - we should delete it when we merge repositories
+
+require 'gooddata'
+require 'json'
+require_relative 'constants'
+
+module GoodData
+  module Helpers
+    # Creates a testing project and loads it with data
+    class AppstoreProjectHelper
+      attr_reader :project, :client
+
+      class << self
+        def create(opts = { client: GoodData.connection })
+          new_project = opts[:client].create_project(opts)
+          AppstoreProjectHelper.new(new_project, opts)
+        end
+
+        def suffix
+          hostname = Socket.gethostname
+          timestamp = DateTime.now.strftime('%Y%m%d%H%M%S')
+          suffix = "#{hostname}_#{timestamp}"
+          segment_name_forbidden_chars = /[^a-zA-Z0-9_\\-]+/
+          suffix.scan(segment_name_forbidden_chars).each do |forbidden_characters|
+            suffix.gsub!(forbidden_characters, '_')
+          end
+          suffix
+        end
+      end
+
+      def initialize(project, opts = { client: GoodData.connection })
+        @client = opts[:client]
+        @project = project if project.instance_of? GoodData::Project
+        @project = client.projects(project) if project.instance_of? String
+        GoodData.logger.info "Project ID: #{@project.obj_id}"
+        raise 'ProjectHelper::initialize the parameter "project" '\
```

Comment:
Style/MultilineIfModifier: Favor a normal unless-statement over a modifier clause in a multiline statement.